src/main/bash/sdkman-env.sh

```diff
@@ -34,6 +36,64 @@ function __sdk_env() {
 		return 1
 	fi
 
+	__sdkman_env_each_candidate "$sdkmanrc" "__sdk_use" && 
+		SDKMAN_ENV=$PWD
+}
+
+function __sdkman_create_env_file() {
+	if [[ -f "$sdkmanrc" ]]; then
```

Comment:
I would prefer to explicitly pass the `sdkmanrc` variable as a parameter. 

Relying on dynamic scoping makes the code harder to understand because I now also need to take the calling function into account.

@marc0der Wdyt?

Reply:
@helpermethod Totally agree, let's not rely on global scope unless absolutely necessary.

Reply:
@marc0der Indeed has dynamic scoping similar to Perl, so sth like this works

```bash
x=10

f() {
  local x=20
  g
}

g() {
  echo "$x"
}

f # prints 20
```

But this is such an obscure feature that I would argue against using it (I've basically learned today that this is possible).

Reply:
TBH I didn't know about dynamic scopes until a couple of days ago 😅  
I thought that maybe in a small module like this one that could be cleaner than passing around the argument to every function.
Anyway it's fixed here 108b03f

Reply:
Very interesting, I wasn't aware of this either. I'd prefer if we just pass values explicitly. Makes the code much easier to follow.